lambdas/handlers/edge_presign_handler.py

```diff
@@ -13,44 +13,81 @@
 logger = logging.getLogger()
 logger.setLevel(logging.INFO)
 
+REQUIRED_QUERY_PARAMS = [
+    "X-Amz-Algorithm",
+    "X-Amz-Credential",
+    "X-Amz-Date",
+    "X-Amz-Expires",
+    "X-Amz-SignedHeaders",
+    "X-Amz-Signature",
+    "X-Amz-Security-Token",
+]
+
+REQUIRED_HEADERS = ["host", "cloudfront-viewer-country", "x-forwarded-for"]
+
 
 @set_request_context_for_logging
 @override_error_check
 @handle_edge_exceptions
 def lambda_handler(event, context):
     try:
         request: dict = event["Records"][0]["cf"]["request"]
-        logger.info("CloudFront received S3 request", {"Result": {json.dumps(request)}})
-        uri: str = request.get("uri", "")
-        presign_query_string: str = request.get("querystring", "")
+        logger.info(f"CloudFront received S3 request {json.dumps(request)}")
+        uri: str = request["uri"]
+        querystring: str = request["querystring"]
+        headers: dict = request["headers"]
+        try:
+            origin = request.get("origin", {})
+            domain_name = origin["s3"]["domainName"]
+        except KeyError as e:
+            logger.error(
+                f"Missing origin: {str(e)}",
+                {"Result": LambdaError.EdgeNoOrigin.to_str()},
+            )
+            raise CloudFrontEdgeException(500, LambdaError.EdgeNoOrigin)
 
-    except (KeyError, IndexError) as e:
-        logger.error(
-            f"{str(e)}",
-            {"Result": {LambdaError.EdgeMalformed.to_str()}},
-        )
-        raise CloudFrontEdgeException(500, LambdaError.EdgeMalformed)
+        try:
+            query_params = {k: v[0] for k, v in parse_qs(querystring).items()}
```

Comment:
Could we add proper variable names instead of `k` and `v` so it's clear what we're accessing?